File: LeetCode/Easy/Apply Operations To An Array/Go/apply_operations_to_array.go
package main

import "fmt"

func apply_operations_to_array(arr []int) []int {
	n := len(arr)
	for i := 0; i < n-1; i++ {
		if arr[i] == arr[i+1] {
			arr[i] *= 2
			arr[i+1] = 0
		}
	}
	i, j := 0, 0
	for i < n && j < n {
		if arr[j] != 0 {
			arr[i] = arr[j]
			i++
			j++
		} else {
			j++
		}
	}
	for k := i; k < n; k++ {
		arr[k] = 0
	}
	return arr
}

func main() {
	fmt.Println(apply_operations_to_array([]int{1, 2, 2, 1, 1, 0}))
}

File: LeetCode/Easy/Apply Operations To An Array/Cpp/apply_operations_to_array.cc
#include <iostream>
#include <vector>

typedef std::vector<int> vi;

template<typename T>
std::ostream& operator<<(std::ostream& os, const std::vector<T>& v) {
    os<<"[";
    for (int i=0;i<v.size();i++) {
        os<<v[i]<<" ";
    }
    os<<"]\n";
    return os;
}

vi apply_operations_to_array(vi v) {
    for (int i=0; i<v.size()-1; i++) {
        if (v[i] == v[i+1]) {
            v[i] *= 2;
            v[i+1] = 0;
        }
    }
    int i=0, j=0;
    while(i<v.size() && j<v.size()) {
        if (v[j] != 0) {
            v[i] = v[j];
            i++;
            j++;
        }else {
            j++;    
        }
    }
    for (int k=i;k<v.size();k++){
        v[k] = 0;
    }
    return v;
}

int main() {
    std::cout<<apply_operations_to_array(vi{1,2,2,1,1,0});
}
